File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {

        int majority = -1;
        int count = 0;

        for(int i = 0; i < nums.length; ++i) {
            if(count == 0) {
                majority = nums[i];
                count++;
            } else if(majority == nums[i]) {
                count++;
            } else {
                count--;
            }
        }

        return majority;



        // int majorityElement = 0;
        // int n = nums.length;

        // Map<Integer, Integer> map = new HashMap<>();

        // for (int i = 0; i < n; ++i) {
        //     map.merge(nums[i], 1, Integer::sum);
        // }

        // int halfN = n / 2;
        // for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
        //     if (entry.getValue() > halfN) {
        //         majorityElement = entry.getKey();
        //     }
        // }

        // return majorityElement;
    }
}

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        count = 0
        element = -1

        for i in range(len(nums)):
            if count == 0:
                count = 1
                element = nums[i]
            elif element == nums[i]:
                count += 1
            else:
                count -= 1

        count = 0
        for i in range(len(nums)):
            if nums[i] == element:
                count += 1

        if count > (len(nums) / 2):
            return element
        return -1



        